tutorials/managing_oli_credentials.ipynb

```diff
@@ -0,0 +1,217 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "e2b807a5",
+   "metadata": {},
+   "source": [
+    "### This notebook provides a brief tutorial on managing OLI Cloud credentials and access keys.\n",
+    "\n",
+    "source code: `watertap.tools.oliapi.credentials.py`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eb86044e",
+   "metadata": {},
+   "source": [
+    "A `CredentialManager` class object is used to manage OLI credentials. This object accepts user credentials which include:\n",
+    "- `username`: e-mail address\n",
+    "- `password`: OLI Cloud password\n",
+    "- `root_url`: url for the API\n",
+    "- `auth_url`: url for API authentication protocols\n",
+    "\n",
+    "The `root_url` and `auth_url` are fixed for all users but should be kept private.\n",
+    "\n",
+    "####  There are 3 ways to log into OLI Cloud:\n",
+    "1) Provide `username`, `password`, `root_url`, and `auth_url`\n",
+    "2) Provide `access_keys` and `root_url`\n",
+    "3) Use `encryption_key` and saved credentials\n",
+    "\n",
+    "Each of these methods will be shown.\n",
+    "\n",
+    "Generating and deleting access keys will also be shown."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b11922c6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from watertap.tools.oli_api.credentials import CredentialManager"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8295c096",
+   "metadata": {},
+   "source": [
+    "#### 1) On first login, you need to provide your username and password\n",
+    "\n",
+    "`CredentialManager` will prompt you to save your `credentials` in an encrypted local file, and provide you an encryption key to log in with."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "752c3bef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    credential_manager = CredentialManager(\n",
+    "        #encryption_key=\"\",\n",
```

Comment:
Is it necessary to include `encryption_key=""`? Can I just not include the argument, resulting in either the default being set to empty string or, if a local file with keys is found, prompt user for encryption key? My main question is do I have to include the line `encryption_key=""` or did you just include for clarity?

Reply:
It is not necessary to include `encryption_key=""` in the arguments. Each argument is listed for all login scenarios in the example, and when an argument is not required for a scenario I've commented it out.